contracts/cw721-base/src/execute.rs

```diff
@@ -125,6 +123,39 @@ where
             .add_attribute("owner", owner)
             .add_attribute("token_id", token_id))
     }
+
+    pub fn update_ownership(
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        action: cw_ownable::Action,
+    ) -> Result<Response<C>, ContractError> {
+        let ownership = cw_ownable::update_ownership(deps, &env.block, &info.sender, action)?;
+        Ok(Response::new().add_attributes(ownership.into_attributes()))
+    }
+
+    pub fn migrate(deps: DepsMut, msg: MigrateMsg) -> Result<Response<C>, ContractError> {
+        let ContractVersion { version, .. } = get_contract_version(deps.storage)?;
+        set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+        match msg {
+            MigrateMsg::From016 {} => {
+                if version != "0.16.0" {
+                    Err(ContractError::WrongMigrateVersion(version))
+                } else {
+                    use cw721_base_016 as v16;
+                    let tract16 = v16::Cw721Contract::<T, C, E, Q>::default();
+                    let minter = tract16.minter.load(deps.storage)?;
+                    tract16.minter.remove(deps.storage);
+                    cw_ownable::initialize_owner(deps.storage, deps.api, Some(minter.as_str()))?;
+                    let ownership = cw_ownable::get_ownership(deps.storage)?;
+                    Ok(Response::new()
+                        .add_attribute("method", "migrate_from_016")
```

Comment:
Other execute method using `"action"` has the key here. I think we should be consistent here. Also suggest just naming this action as "migrate" and use attributes to indicate the specific migrations being migrated from/to:

```rust
Response::new()
    .add_attribute("action", "migrate")
    .add_attribute("from_version", "0.16")
    .add_attribute("to_version", "0.17")
```